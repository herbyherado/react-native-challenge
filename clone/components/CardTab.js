import React, { Component } from 'react'
import { Text, View, TouchableHighlight, StyleSheet } from 'react-native' 

class CardTab extends Component {
  render() {
    return (
      <TouchableHighlight onPress = { this.props.press }>
        <View style={styles.card}> 
          <Text style={styles.cardText} > { this.props.text }</Text>
        </View>
      </TouchableHighlight>  
    )
  }
};

const styles = StyleSheet.create({
  card: {
    marginTop: 20,
    alignItems: 'center',
    alignSelf: 'center',
    padding: 30,
    width: '95%',
    borderWidth: 1,
    borderBottomColor: 'silver',
    borderTopColor: 'silver'
  },
  cardText: {
    color: '#fff',
    fontSize: 30,
    backgroundColor: 'transparent'
  },
})

export default CardTab