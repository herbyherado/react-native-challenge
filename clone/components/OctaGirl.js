import React, { Component } from 'react';
import { View, Text, Image, StyleSheet } from 'react-native'

class OctaGirl extends Component {
  render() {
    return (
      <View>
        <Image style={styles.img} source={{ uri: this.props.item.medium_profile_picture }}/>
        <Text>{`${this.props.item.first_name} ${this.props.item.last_name}`}</Text>
        <Text>{ this.props.item.country_residing }</Text>
        <Text>{ this.props.item.quote }</Text>
      </View>
    )
  }
};

const styles = StyleSheet.create({
  img: {
    width: 150,
    height: 150,
    alignSelf: 'center'
  }
})

export default OctaGirl