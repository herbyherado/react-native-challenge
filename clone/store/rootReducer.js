import { combineReducers } from 'redux'
import Fighter from './fighters/fighter.reducer'
import Octagirl from './octagirls/octagirls.reducer'

const reducers = combineReducers({
  fighters: Fighter,
  octagirls: Octagirl
})

export default reducers