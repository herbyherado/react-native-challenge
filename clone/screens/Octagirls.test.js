import React, { Component } from 'react'
import enzyme, { shallow, mount } from 'enzyme'
import { Text, View, Image, FlatList } from 'react-native'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import OctaGirls from './Octagirls'
import OctaGirl from '../components/OctaGirl';
import CardTab from '../components/CardTab'
import store from '../store/store'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()
let initialState = {
  octagirls: []
}
let mstore
let wrapper

enzyme.configure({ adapter: new Adapter()})

beforeEach(() => {
  mstore = mockStore(initialState)
  wrapper = shallow(<Provider store={store}><OctaGirls/></Provider>)
})

// describe('Snapshot test <OctaGirls/>',()=>{
//   it('capturing Snapshot of OctaGirls', () => {
//       const renderedValue =  renderer.create(<Provider store={store}><OctaGirls/></Provider>).toJSON()
//       expect(renderedValue).toMatchSnapshot();
//   })
// })

describe('<OctaGirls/>', () => {
  it('should render <OctaGirls/>', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper.length).toEqual(1)
  })
  it('should have components', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('Text')).toBeDefined()
    expect(wrapper.find('OctaGirl')).toBeDefined()
    expect(wrapper.find('FlatList')).toBeDefined()
  })
  it('should have props', () => {
    expect(wrapper.props('octagirls')).toEqual({})
  })
  // it('keyextractor', () => {
    expect(wrapper.dive().instance()).toEqual('y')
    // console.log(wrapper.instance().keyExtractor({id: '113'}, '4'))
    // let key = wrapper.instance().keyExtractor({id: '113'}, '4')
    // expect(key).toEqual('octa-113')
    // expect(OctaGirls.keyExt)
  // })
})

