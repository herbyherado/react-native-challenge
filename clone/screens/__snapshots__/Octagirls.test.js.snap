// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test <OctaGirls/> capturing Snapshot of OctaGirls 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 2,
    }
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
